controlplane/kubeadm/internal/control_plane.go

```diff
@@ -385,3 +389,79 @@ func (c *ControlPlane) InjectTestManagementCluster(managementCluster ManagementC
 	c.managementCluster = managementCluster
 	c.workloadCluster = nil
 }
+
+// StatusToLogKeyAndValues returns following key/value pairs describing the overall status of the control plane:
+// - machines is the list of KCP machines; each machine might have additional notes surfacing
+//   - if the machine has been created in the current reconcile (new)
+//   - if machines node is not yet (node ref not set)
+//   - if the machine has bee marked for remediation (health check failed)
+//   - if there are unhealthy control plane component on the machine
+//   - if the machine has a deletion timestamp/has been deleted in the current reconcile (deleting)
+//   - if the machine is not up to date with the KCP spec (not up to date)
+//
+// - etcdMembers list as reported by etcd.
+func (c *ControlPlane) StatusToLogKeyAndValues(newMachine, deletedMachine *clusterv1.Machine) []any {
+	controlPlaneMachineHealthConditions := []clusterv1.ConditionType{
+		controlplanev1.MachineAPIServerPodHealthyCondition,
+		controlplanev1.MachineControllerManagerPodHealthyCondition,
+		controlplanev1.MachineSchedulerPodHealthyCondition,
+	}
+	if c.IsEtcdManaged() {
+		controlPlaneMachineHealthConditions = append(controlPlaneMachineHealthConditions,
+			controlplanev1.MachineEtcdPodHealthyCondition,
+			controlplanev1.MachineEtcdMemberHealthyCondition,
+		)
+	}
+
+	machines := []string{}
+	for _, m := range c.Machines {
+		notes := []string{}
+
+		if m.Status.NodeRef == nil {
+			notes = append(notes, "node ref not set")
+		}
+
+		if c.MachinesToBeRemediatedByKCP().Has(m) {
+			notes = append(notes, "health check failed")
+		}
+
+		for _, condition := range controlPlaneMachineHealthConditions {
+			if conditions.IsUnknown(m, condition) {
+				notes = append(notes, strings.Replace(string(condition), "Healthy", " health unknown", -1))
+			}
+			if conditions.IsFalse(m, condition) {
+				notes = append(notes, strings.Replace(string(condition), "Healthy", " not healthy", -1))
+			}
+		}
+
+		if !c.UpToDateMachines().Has(m) {
```

Comment:
Wondering if we should also set "up-to-date" in the else case

Reply:
I would prefer not to kepp the message as concise as possible